File: hashMap/containsDuplicate.js
const containsDuplicate = (nums) => {
  const seen =  new Set();
  for (let num of nums) {
    if(seen.has(num)) return true;
    seen.add(num)
  }
  return false;
}

console.assert(containsDuplicate([1,2,3,1]));
console.assert(containsDuplicate([1,2,3,4]) === false, 'returns false no dupe') ;
console.assert(containsDuplicate([1,1,1,1,3,3]), 'returns true') ;
console.assert(containsDuplicate([4,1,2,5,3,4]), 'returns true') ;

File: arrays/containsDuplicate.py
class Solution:
  def containsDuplicate(self, nums: List[int]) -> bool:
    seen = set()
    for num in nums:
      if(num in seen) return num
    return len(nums) !== len(set(nums))

    # Create hash map or set

    # Add the number to the set if you havent seen this number

    # If you seen the number return that number